src/robot_model_builder.cpp

```diff
@@ -0,0 +1,175 @@
+       it != pinocchio_model_complete.names.end(); ++it) {
+    const std::string& joint_name = *it;
+    if (std::find(moving_joint_names_.begin(), moving_joint_names_.end(),
+                  joint_name) == moving_joint_names_.end()) {
+      locked_joint_ids_.push_back(
+          pinocchio_model_complete.getJointId(joint_name));
+    }
+  }
+
+  // remove the "root_joint" if it exists in the moving_joint_names_, we needed
+  // it to create the above locked_joint_ids needed for the reduced model.
+  auto root_name = std::remove(moving_joint_names_.begin(),
+                               moving_joint_names_.end(), "root_joint");
+  if (root_name != moving_joint_names_.end()) {
+    moving_joint_names_.erase(root_name, moving_joint_names_.end());
+    auto root_id =
+        std::remove(moving_joint_ids_.begin(), moving_joint_ids_.end(),
+                    pinocchio_model_complete.getJointId("root_joint"));
+    moving_joint_ids_.erase(root_id, moving_joint_ids_.end());
+  }
+
+  // Create the joint name joint id mapping of the hardware interface.
+  for (std::size_t i = 0; i < moving_joint_names_.size(); ++i) {
+    auto it = std::find(controlled_joint_names.begin(),
+                        controlled_joint_names.end(), moving_joint_names_[i]);
+    // verify that the found moving joints are part of the hardware interface.
+    if (it == controlled_joint_names.end()) {
+      std::cerr << "LinearFeedbackController::parseMovingJointNames(): "
+                << "Moving joint " << moving_joint_names_[i]
+                << " is not part of the current roscontrol hardware "
+                << "interface." << std::endl;
+      return false;
+    }
+    std::size_t it_dist =
+        (size_t)std::distance(controlled_joint_names.begin(), it);
+    pin_to_hwi_.emplace(std::make_pair(i, it_dist));
+  }
+  // Double check what we just did
+  for (std::size_t i = 0; i < moving_joint_names_.size(); ++i) {
+    if (moving_joint_names_[i] != controlled_joint_names[pin_to_hwi_[i]]) {
+      std::cerr << "LinearFeedbackController::parseMovingJointNames(): "
+                << "The mapping from pinocchio to roscontrol is wrong. "
+                << "moving_joint_names_[i] != "
+                << "controlled_joint_names[pin_to_hwi_[i]] => "
+                << moving_joint_names_[i]
+                << " != " << controlled_joint_names[pin_to_hwi_[i]];
+      return false;
+    }
+  }
+  return true;
+}
+
+const std::vector<std::string>& RobotModelBuilder::get_moving_joint_names()
+    const {
+  return moving_joint_names_;
+}
+
+const std::vector<long unsigned int>& RobotModelBuilder::get_moving_joint_ids()
+    const {
+  return moving_joint_ids_;
+}
+
+const std::vector<long unsigned int>& RobotModelBuilder::get_locked_joint_ids()
+    const {
+  return locked_joint_ids_;
+}
+
+const pinocchio::Model& RobotModelBuilder::get_model() const {
+  return pinocchio_model_;
+}
+
+pinocchio::Data& RobotModelBuilder::get_data() { return pinocchio_data_; }
```

Comment:
Did you forget a const or not ?

Reply:
No the idea is to use the data in the computation.